pyanaconda/network.py

```diff
@@ -318,13 +319,17 @@ def write_configuration(overwrite=False):
     task_proxy = NETWORK.get_proxy(task_path)
     sync_run_task(task_proxy)
 
-    task_path = network_proxy.InstallNetworkWithTask(overwrite)
-    task_proxy = NETWORK.get_proxy(task_path)
-    sync_run_task(task_proxy)
-
-    task_path = network_proxy.ConfigureHostnameWithTask(overwrite)
-    task_proxy = NETWORK.get_proxy(task_path)
-    sync_run_task(task_proxy)
+    if is_module_available(NETWORK) \
+            or network_proxy.Kickstarted:
```

Comment:
I think we shouldn't disable modules that we still plan to use in some cases. This wasn't the intended use case. It is just a nice fallback that prevents anaconda from crashing if we forget to handle the disabled module somewhere. In the future, it shouldn't be allowed at all.

It looks like these cases could be configurable via our configuration files.